sources/packages/backend/apps/workers/src/services/application-exception/application-exception-hash.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from "@nestjs/common";
+import { In, Repository } from "typeorm";
+import { StudentFile } from "@sims/sims-db";
+import { InjectRepository } from "@nestjs/typeorm";
+import {
+  ApplicationDataException,
+  ApplicationDataExceptionHashed,
+} from "./application-exception.models";
+import { hashObjectToHex } from "@sims/utilities";
+
+/**
+ * Process {@see ApplicationDataException} to create a hash
+ * that represents the exception data content and its files.
+ */
+@Injectable()
+export class ApplicationExceptionHashService {
+  constructor(
+    @InjectRepository(StudentFile)
+    private readonly studentFileRepo: Repository<StudentFile>,
+  ) {}
+
+  /**
+   * Creates hashed representations of application exceptions.
+   * @param exceptions application exceptions to hash.
+   * @param studentId ID of the student associated with the exceptions.
+   * Used for authorization purposes to ensure the user has access to the files.
+   * @returns hashed application exceptions.
+   */
+  async createHashedApplicationExceptions(
+    exceptions: ApplicationDataException[],
+    studentId: number,
+  ): Promise<ApplicationDataExceptionHashed[]> {
+    const filesHashMap = await this.getFilesHashMapForExceptions(
+      exceptions,
+      studentId,
+    );
+    return exceptions.map((exception) =>
+      this.hashContent(exception, filesHashMap),
+    );
+  }
+
+  /**
+   * Hashes the content of an application exception.
+   * @param exception application exception to hash.
+   * @param filesHashMap map of file names to their hash values.
+   * @returns hashed application exception.
+   */
+  private hashContent(
+    exception: ApplicationDataException,
+    filesHashMap: Record<string, string>,
+  ): ApplicationDataExceptionHashed {
+    // Combine all file hashes in a sorted manner to ensure consistent ordering.
+    exception.files.sort((a, b) => a.name.localeCompare(b.name));
+    const filesHashes = exception.files.map((file) => {
+      const fileHash = filesHashMap[file.name];
+      if (!fileHash) {
+        throw new Error(
+          `File hash not found for file: ${file.name}. Ensure the file exists in the database.`,
+        );
+      }
+      return fileHash;
+    });
+    // Combine the exception data content and its files hashes to create a full hash.
+    const fullStringContent =
+      JSON.stringify(exception.hashableContent) + filesHashes.join();
```

Comment:
We generally use string template to join the strings. 

```ts

const fullStringContent = `${JSON.stringify(
      exception.hashableContent,
    )}${filesHashes.join()}`;
```

Reply:
Changed.